File: src/main/java/org/opensearch/commons/ppl/TransportPPLQueryResponse.java
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

package org.opensearch.commons.ppl;

import java.io.IOException;

import org.opensearch.core.action.ActionResponse;
import org.opensearch.core.common.io.stream.StreamInput;
import org.opensearch.core.common.io.stream.StreamOutput;

/**
 * This is a copied, reduced version of SQL Plugin's TransportPPLQueryResponse
 */
public class TransportPPLQueryResponse extends ActionResponse {

    private static final String CONTENT_TYPE = "application/json; charset=UTF-8";

    private final String result;
    private final String contentType;

    public TransportPPLQueryResponse(String result) {
        this.result = result;
        this.contentType = CONTENT_TYPE;
    }

    public TransportPPLQueryResponse(StreamInput in) throws IOException {
        super(in);
        result = in.readString();
        contentType = in.readString();
    }

    @Override
    public void writeTo(StreamOutput out) throws IOException {
        // super.writeTo(out) is a no-op, matching the SQL plugin's implementation
        out.writeString(result);
        out.writeString(CONTENT_TYPE);
    }

    public String getResult() {
        return result;
    }

    public String getContentType() {
        return contentType;
    }
}

File: src/main/kotlin/org/opensearch/commons/sql/action/TransportPPLQueryResponse.kt
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

package org.opensearch.commons.sql.action

import org.opensearch.common.io.stream.StreamInput
import org.opensearch.common.io.stream.StreamOutput
import org.opensearch.common.io.stream.Writeable
import org.opensearch.common.xcontent.ToXContent
import org.opensearch.common.xcontent.XContentBuilder
import java.io.IOException

/**
 * Action Response for ppl query.
 */
class TransportPPLQueryResponse : BaseResponse {
    val queryResponse: String

    companion object {
        /**
         * reader to create instance of class from writable.
         */
        val reader = Writeable.Reader { TransportPPLQueryResponse(it) }
    }

    /**
     * constructor for creating the class
     * @param queryResponse the ppl query response
     */
    constructor(
        queryResponse: String,
    ) {
        this.queryResponse = queryResponse
    }

    /**
     * {@inheritDoc}
     */
    @Throws(IOException::class)
    constructor(input: StreamInput) : super(input) {
        queryResponse = input.readString()
    }

    @Throws(IOException::class)
    override fun writeTo(output: StreamOutput) {
        output.writeString(queryResponse)
    }

    /**
     * {@inheritDoc}
     */
    override fun toXContent(builder: XContentBuilder?, params: ToXContent.Params?): XContentBuilder {
        throw IllegalStateException("Transport ppl response is not intended for REST or persistence and does not support XContent.")
    }
}
